src/braket/default_simulator/openqasm/program_context.py

```diff
@@ -113,29 +115,73 @@ def _(self, identifier: IndexedIdentifier) -> tuple[int]:
         corresponding to the elements referenced by the indexed identifier.
         """
         name = identifier.name.name
+        indices = self.get_qubit_indices(identifier)
+        primary_index = indices[0]
+
+        if isinstance(primary_index, (IntegerLiteral, SymbolLiteral)):
+            target = (self[name][0] + primary_index.value,)
+        elif isinstance(primary_index, RangeDefinition):
+            target = tuple(np.array(self[name])[convert_range_def_to_slice(primary_index)])
+        # Discrete set
+        else:
+            index_list = convert_discrete_set_to_list(primary_index)
+            target = tuple([self[name][0] + index for index in index_list])
+
+        if len(indices) == 2:
+            # used for gate calls on registers, index will be IntegerLiteral
+            secondary_index = indices[1].value
+            target = (target[secondary_index],)
+
+        # validate indices manually, since we use addition instead of indexing to
+        # accommodate symbolic indices
```

Comment:
This function is basically the same as before, with the `[]` -> `+` changes and this extra validation (that we lose from not using indexing). Technically the error we get now is better anyways